src/base/object_representation.h

```diff
@@ -610,10 +634,10 @@
 #define OR_MVCC_INSERT_ID_OFFSET (OR_CHN_OFFSET + OR_CHN_SIZE)
 #define OR_MVCC_INSERT_ID_SIZE 8
 
-#define OR_MVCC_DELETE_ID_OFFSET (OR_MVCC_INSERT_ID_OFFSET + OR_MVCC_INSERT_ID_SIZE)
+#define OR_MVCC_DELETE_ID_OFFSET(mvcc_flags) (OR_MVCC_INSERT_ID_OFFSET + (((mvcc_flags) & OR_MVCC_FLAG_VALID_INSID) ? OR_MVCC_INSERT_ID_SIZE : 0))
 #define OR_MVCC_DELETE_ID_SIZE 8
 
-#define OR_MVCC_PREV_VERSION_LSA_OFFSET (OR_MVCC_DELETE_ID_OFFSET + OR_MVCC_DELETE_ID_SIZE)
+#define OR_MVCC_PREV_VERSION_LSA_OFFSET(mvcc_flags) (OR_MVCC_DELETE_ID_OFFSET + (((mvcc_flags) & OR_MVCC_FLAG_VALID_DELID) ? OR_MVCC_DELETE_ID_SIZE : 0))
```

Comment:
You have to use `OR_MVCC_DELETE_ID_OFFSET `with the mvcc_flags argument.

Make sure the macro don't have more than 120 charactes per line. Automatic indent don't work here.